src/TestableIO.System.IO.Abstractions.TestingHelpers/Events/MockFileSystemEvents.cs

```diff
@@ -0,0 +1,394 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace System.IO.Abstractions.TestingHelpers.Events;
+
+/// <summary>
+/// Provides event functionality for MockFileSystem operations.
+/// </summary>
+/// <example>
+/// <code>
+/// // Subscribe to all operations
+/// var fs = new MockFileSystem(new MockFileSystemOptions { EnableEvents = true });
+/// var subscription = fs.Events.Subscribe(args => 
+/// {
+///     Console.WriteLine($"{args.Operation} {args.Path}");
+/// });
+/// 
+/// // Subscribe to specific operations
+/// var writeSub = fs.Events.Subscribe(FileOperation.Write, args =>
+/// {
+///     if (args.Phase == OperationPhase.Before)
+///     {
+///         args.SetResponse(new OperationResponse 
+///         { 
+///             Exception = new IOException("Disk full") 
+///         });
+///     }
+/// });
+/// 
+/// // Unsubscribe when done
+/// subscription.Dispose();
+/// </code>
+/// </example>
+#if FEATURE_SERIALIZABLE
+[Serializable]
+#endif
+public class MockFileSystemEvents
+{
+#if FEATURE_SERIALIZABLE
+    /// <summary>
+    /// Represents the collection of active event subscriptions for handling file system operations.
+    /// </summary>
+    [NonSerialized]
+#endif
+    private readonly List<Subscription> subscriptions = [];
```

Comment:
You could use a `ConcurrentDictionary` (or another thread-safe collection), to avoid using the `LockObject`...